client/asset/btc/btc.go

```diff
@@ -3343,8 +3364,11 @@ func (btc *baseWallet) createSig(tx *wire.MsgTx, idx int, pkScript []byte, addr
 // input and the pubkey associated with the address.
 func (btc *baseWallet) createWitnessSig(tx *wire.MsgTx, idx int, pkScript []byte,
 	addr btcutil.Address, val uint64, sigHashes *txscript.TxSigHashes) (sig, pubkey []byte, err error) {
-
-	privKey, err := btc.node.privKeyForAddress(addr.String())
+	addrStr, err := btc.stringAddr(addr, btc.chainParams)
+	if err != nil {
+		return nil, nil, err
+	}
+	privKey, err := btc.node.privKeyForAddress(addrStr)
```

Comment:
Not sure, but perhaps `bitcoincashd` (Bitcoin Cash Node)` was able to pull out the pubkey from the bitcoin addresses and locate the private key.  The chain params were the clone params with the correct bch address ID values, but this wasn't making Cash Addresses.

Reply:
Yeah, that's what it was doing.

```
$  ./beta getaddressinfo mgDzcoxdBrhu435knkdUEP5UGVYGKUJJgN
{
  "address": "bchreg:qqru9lns98ugq85h4cl3p8srf4y9w8cyt5eux55xmn",
  "scriptPubKey": "76a91407c2fe7029f8801e97ae3f109e034d48571f045d88ac",
  "ismine": true,
  "iswatchonly": false,
  "isscript": false,
  "pubkey": "0310892969dde0c7f6e177ce0f1a8ba807fbbbbc05135e9c47479f0aa1fa2555a5",
...
```

Reply:
Interesting.